### PR TITLE
2: Hide synced spaces in spaces browser

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #11752

Depends on https://github.com/owncloud/client/pull/11758

### CODE REVIEW

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -130,6 +123,9 @@ ecm_add_qml_module(owncloudGui
             qml/AccountButton.qml
             qml/FolderDelegate.qml
             qml/FolderError.qml
+
+            spaces/qml/SpaceDelegate.qml
```
Comment (TheOneRing) [inline src/gui/CMakeLists.txt:127]: Adding the qml file in the subdirectory failed during the cache generation so I had to move it here.

File: src/gui/spaces/CMakeLists.txt [hunk 2]
```diff
@@ -1,25 +1,7 @@
-add_library(ownCloudSpaces SHARED
+target_sources(owncloudGui
```
Comment (TheOneRing) [inline src/gui/spaces/CMakeLists.txt:1]: I finally had to merge the spaces browser into the gui module....
In order to track which spaces are synced I needed to access the folder manager.....

File: changelog/unreleased/11752 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CHange: Only allow to add unsyced spaces
```
Comment (erikjv) [inline changelog/unreleased/11752:1]: The second letter shouldn't be capitalised

File: src/gui/folderman.cpp [hunk 4]
```diff
@@ -909,6 +909,12 @@ Result<void, QString> FolderMan::unsupportedConfiguration(const QString &path) c
     return *it;
 }
 
+bool FolderMan::isSpaceSynced(GraphApi::Space *space) const
```
Comment (erikjv) [inline src/gui/folderman.cpp:912]: This is already in another PR, so it needs a rebase?

**Review by erikjv — APPROVED** [review]

**Comment by TheOneRing** [comment]
@saw-jan this changes breaks a gui test as your synced the same space multiple times.

**Review by erikjv — APPROVED** [review]

**Comment by saw-jan** [comment]
> @saw-jan this changes breaks a gui test as your synced the same space multiple times.

Will have a look. Thanks

**Pull request merged**